### PR TITLE
Mikaelm12/originalValueHeaders

### PR DESCRIPTION
**Pull request by mikaelm12**
Storing original IP, host, and protocol values in new headers.
@victorhurdugaci  @Tratcher @blowdart

### CODE REVIEW

File: src/Microsoft.AspNet.HttpOverrides/OverrideHeaderMiddleware.cs [hunk 1]
```diff
@@ -31,21 +34,27 @@ public Task Invoke(HttpContext context)
                 var xForwardedForHeaderValue = context.Request.Headers.GetCommaSeparatedValues(XForwardedForHeaderName);
                 if (xForwardedForHeaderValue != null && xForwardedForHeaderValue.Count > 0)
                 {
-                    IPAddress originalIPAddress;
-                    if (IPAddress.TryParse(xForwardedForHeaderValue[0], out originalIPAddress))
+                    IPAddress IPFromHeader;
```
Comment (Tratcher) [inline src/Microsoft.AspNet.HttpOverrides/OverrideHeaderMiddleware.cs]: nit: lower case variable name ipFromHeader

File: src/Microsoft.AspNet.HttpOverrides/OverrideHeaderMiddleware.cs [hunk 2]
```diff
@@ -31,21 +34,27 @@ public Task Invoke(HttpContext context)
...
                     }
                 }
             }
 
             if ((_options.ForwardedOptions & ForwardedHeaders.XForwardedHost) != 0)
             {
+                var hostString = context.Request.Host.ToString();
+                if (!string.IsNullOrEmpty(hostString))
+                {
+                    context.Request.Headers.Set(XOriginalHostName, hostString);
```
Comment (Tratcher) [inline src/Microsoft.AspNet.HttpOverrides/OverrideHeaderMiddleware.cs]: You should only set this if you're going to modify Request.Host. Same for Scheme.

**Comment by mikaelm12** [comment]
Updated

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**